src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -131,8 +134,10 @@ public override void OnCustom(string name, Activity activity, object payload)
                     }
 
                     break;
-                case SqlDataAfterExecuteCommand:
-                case SqlMicrosoftAfterExecuteCommand:
+                case SqlDataAfterExecuteCommand when this.options.ActivityStopTriggerEvent == SqlClientActivityStopTriggerEvent.WriteCommandAfter:
+                case SqlMicrosoftAfterExecuteCommand when this.options.ActivityStopTriggerEvent == SqlClientActivityStopTriggerEvent.WriteCommandAfter:
+                case SqlDataAfterConnectionClose when this.options.ActivityStopTriggerEvent == SqlClientActivityStopTriggerEvent.WriteConnectionCloseAfter:
+                case SqlMicrosoftAfterConnectionClose when this.options.ActivityStopTriggerEvent == SqlClientActivityStopTriggerEvent.WriteConnectionCloseAfter:
```

Comment:
Do you have some example code or tests you could contribute? I'm trying to get my head around when an activity would be ended when using `WriteConnectionCloseAfter`.

One scenario I'm curious about is opening a connection and then running multiple commands:

```csharp
using (var connection = new SqlConnection(...))
{
    connection.Open();
    using (var firstCommand = new SqlCommand("SELECT ...", connection))
    {
    }
    using (var secondCommand = new SqlCommand("SELECT ...", connection))
    {
    }
}
```

Wouldn't this create multiple activities? Would they all get stopped?

Reply:
+1

Reply:
We have this case
`
            using (dbcontext)
            {
                var results = context.SomeDbSet.Select(x => ....).ToArray();
            }`

We noticed that one query that returns a lot of rows (300k+) the actual time until we received the full results was around 16 seconds, but the trace showed a duration of 77ms. The actual duration was recorded if the activity stopped on connection close (disposing the db context).

You are correct though, my implementation could lead to confusing results in some (maybe most) use cases and even though this is offered as a non-default option, I will change this so that it will provide the option to start-stop activity based on connection open/close and keep the default as command exec before/after.